spec/javascripts/surveys-spec.js

```diff
@@ -90,25 +87,109 @@ describe('Surveys', function () {
       expect(GOVUK.cookie(surveys.surveySeenCookieName(defaultSurvey))).toBe('1')
     })
 
-    describe("for a 'url' survey", function () {
-      it('links to the url for a smartsurvey survey with a completion redirect query parameter', function () {
-        surveys.displaySurvey(urlSurvey)
+    describe('for a url survey', function () {
+      it('replaces the current path if the survey url contains the {{currentPath}} template parameter', function () {
+        var urlSurveyWithCurrentPath = {
+          surveyType: 'url',
+          url: 'smartsurvey.com/default?c={{currentPath}}',
+          identifier: 'url-survey'
+        }
+        surveys.displaySurvey(urlSurveyWithCurrentPath)
 
-        expect($('#take-survey').attr('href')).toContain(urlSurvey.url)
         expect($('#take-survey').attr('href')).toContain('?c=' + window.location.pathname)
       })
 
-      it('sets event handlers on the survey', function () {
-        spyOn(surveys, 'setURLSurveyEventHandlers')
+      it('does not inject the current path if the survey url does not contain the {{currentPath}} template parameter', function () {
         surveys.displaySurvey(urlSurvey)
-        expect(surveys.setURLSurveyEventHandlers).toHaveBeenCalledWith(urlSurvey)
+
+        expect($('#take-survey').attr('href')).toEqual(urlSurvey.url)
+      })
+
+      it('injects the url for a non-smartsurvey survey and adds current path as a `c` param', function () {
```

Comment:
I don't think we need this test anymore, as it's a duplicate of the `'replaces the current path if the survey url contains the {{currentPath}} template parameter'` one above.  We no longer have special treatment of smartsurvey urls, so referring to non-smartsurvey urls isn't necessary.